ha/util/consul_kv_store.py

```diff
@@ -154,13 +195,31 @@ def get(self, key: str = ""):
         Return:
             str: Return dictionary of all key val pair.
         """
-        _, data = self._consul.kv.get(self._prepare_key(key), recurse=True)
-        if data is None:
-            return data
+        k = self._prepare_key(key)
+
+        # simulating consul kv get operation for _payload dict
+        payload_data = None
+        if self._enable_batch_put:
+            # Using dictionary comprehension + startswith()
+            # Prefix key match in dictionary
+            Log.debug(f"[consul-op] searching key in cache: {k}")
+            payload_data = {key:val for key, val in self._payload.items() if key.startswith(k)}
+
+        Log.debug(f"[consul-op] getting the value for key: {k}")
+        _, data = self._consul.kv.get(k, recurse=True)
+        Log.debug(f"[consul-op] got the value for Key: {k}")
+
         key_val: dict = {}
-        for key in data:
-            key_val[key['Key']] = key['Value'].decode("utf-8") if isinstance(key['Value'], bytes) else key['Value']
-        return key_val
+        if data:
```

Comment:
Why are we fetching all the keys with a given prefix and not the exact ones?

Reply:
consul.get() works like this when we give some key and recurse=True it finds out all the keys with prefix and I think this functionally is being used here I am added similar functionality that consul.kv.get(key,recurse=True) gives us.

Reply:
Addressed